Scripts/common.ps1

```diff
@@ -0,0 +1,186 @@
+function Test-GitHubToken {
+    if ($Env:GITHUB_TOKEN) {
+        Write-Host 'GITHUB_TOKEN detected'
+        return ${Env:GITHUB_TOKEN}
+    }
+    else {
+        Write-Host 'GITHUB_TOKEN not detected'
+        exit 1
+    }
+}
+
+
+function Test-PackageAndVersionInGithub {
+    param(
+        [Parameter(Mandatory = $true)] [string] $latestVersion,
+        [Parameter(Mandatory = $false)] [string] $wingetPackage = ${Env:PackageName}
+    )
+    Write-Host "Checking if $wingetPackage is already in winget (via GH) and Version $($Latest.Version) already present"
+    $ghVersionURL = "https://raw.githubusercontent.com/microsoft/winget-pkgs/master/manifests/$($wingetPackage.Substring(0, 1).ToLower())/$($wingetPackage.replace(".","/"))/$latestVersion/$wingetPackage.yaml"
+    $ghCheckURL = "https://github.com/microsoft/winget-pkgs/blob/master/manifests/$($wingetPackage.Substring(0, 1).ToLower())/$($wingetPackage.replace(".","/"))/"
+
+    $ghCheck = Invoke-WebRequest -Uri $ghCheckURL -Method Head -SkipHttpErrorCheck 
+    $ghVersionCheck = Invoke-WebRequest -Uri $ghVersionURL -Method Head -SkipHttpErrorCheck
+
+    if ($ghCheck.StatusCode -eq 404) {
+        Write-Host "Packet not yet in winget. Please add new Packet manually"
+        exit 1
+    } 
+    elseif ($ghVersionCheck.StatusCode -eq 200) {
+        Write-Host "Latest version of $wingetPackage $latestVersion is already present in winget."
+        exit 0
+    }
+    else {
+        return $true
+    }
+
+}
+
+function Test-PackageAndVersionInWinget {
+    param(
+        [Parameter(Mandatory = $true)] [string] $latestVersion,
+        [Parameter(Mandatory = $false)] [string] $wingetPackage = ${Env:PackageName}
+    )
+    Write-Host "Checking if $wingetPackage is already in winget and Version $($Latest.Version) already present"
+
+    $progressPreference = 'silentlyContinue'
+    $latestWingetMsixBundleUri = $(Invoke-RestMethod https://api.github.com/repos/microsoft/winget-cli/releases/latest).assets.browser_download_url | Where-Object { $_.EndsWith(".msixbundle") }
+    $latestWingetMsixBundle = $latestWingetMsixBundleUri.Split("/")[-1]
+    Write-Host "Downloading winget to artifacts directory..."
+    Invoke-WebRequest -Uri $latestWingetMsixBundleUri -OutFile "./$latestWingetMsixBundle"
+    Invoke-WebRequest -Uri https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx -OutFile Microsoft.VCLibs.x64.14.00.Desktop.appx
+    Add-AppxPackage Microsoft.VCLibs.x64.14.00.Desktop.appx
+    Add-AppxPackage $latestWingetMsixBundle
+
+    start-sleep 20
```

Comment:
extract winget install in it's own function and ideally also check if install is even necessarry (or if winget is already installed)

Reply:
should not be used anyway ... just there for future switch back